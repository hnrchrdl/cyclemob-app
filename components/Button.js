import React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import { MaterialIcons } from '@expo/vector-icons';

Button.propTypes = {
  onPress: PropTypes.func.isRequired,
  iconName: PropTypes.string.isRequired,
  color: PropTypes.string,
  outline: PropTypes.bool
};

function Button({ iconName, color, onPress, outline }) {
  return (
    <TouchableOpacity
      style={[styles.container, outline ? styles.outlined : {}]}
      onPress={onPress}
    >
      <MaterialIcons style={color ? { color } : {}} name={iconName} size={20} />
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    width: 50,
    height: 50,
    margin: 5,
    backgroundColor: 'transparent'
  },
  outlined: {
    borderWidth: StyleSheet.hairlineWidth,
    borderColor: '#eee',
    borderRadius: 100,
    backgroundColor: '#fff'
  },
  noOutline: {}
});

export default Button;
