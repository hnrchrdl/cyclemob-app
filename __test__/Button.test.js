import React from 'react';
import Button from '../components/Button';

import { shallow } from 'enzyme';

const props = {
  onPress: jest.fn(),
  iconName: 'search'
};
it('renders without crashing', () => {
  const tree = shallow(<Button {...props} />);
  expect(tree).toBeTruthy();
});
it('matches snapshot', () => {
  const tree = shallow(<Button {...props} />);
  expect(tree).toMatchSnapshot();
});
it('handles callback on press', () => {
  const tree = shallow(<Button {...props} />);
  tree.simulate('press');
  expect(props.onPress).toBeCalled();
});
