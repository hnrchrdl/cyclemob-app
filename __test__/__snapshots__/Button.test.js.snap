// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    iconName="search"
    onPress={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <Icon
        allowFontScaling={false}
        name="search"
        size={20}
        style={Object {}}
      />,
      "onPress": [MockFunction],
      "style": Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "height": 50,
          "justifyContent": "center",
          "margin": 5,
          "width": 50,
        },
        Object {},
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowFontScaling": false,
        "name": "search",
        "size": 20,
        "style": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Icon
          allowFontScaling={false}
          name="search"
          size={20}
          style={Object {}}
        />,
        "onPress": [MockFunction],
        "style": Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "height": 50,
            "justifyContent": "center",
            "margin": 5,
            "width": 50,
          },
          Object {},
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": false,
          "name": "search",
          "size": 20,
          "style": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
